### PR TITLE
Remove tags (deprecated) + Update docs for 6.x

### PR DESCRIPTION
**Pull request by basepi**
The docs changes were large enough I wanted to make sure there were no docs build errors before I cut the 6.0.0 release.

### CODE REVIEW

File: docs/upgrading.asciidoc [hunk 1]
```diff
@@ -25,8 +25,22 @@ The table below is a simplified description of this policy.
...
 |5.x |2021-02-05 |6.0
+|6.x |2022-08-01 |7.0
 |====
 
+[[upgrading-6.x]]
+=== Upgrading to version 6 of the agent
+
+==== Python 2 no longer supported
+
+Please upgrade to python 3.6+ to continue to receive regular updates.
```
Comment (beniwohli) [inline docs/upgrading.asciidoc:36]:
```suggestion
Please upgrade to Python 3.6+ to continue to receive regular updates.
```
;-)

File: docs/upgrading.asciidoc [hunk 2]
```diff
@@ -42,7 +56,7 @@ we renamed "tags" to "labels", and introduced limited support for typed labels.
 While tag values were only allowed to be strings, label values can be strings, booleans, or numerical.
 
 To benefit from this change, ensure that you run at least *APM Server 6.7*, and use `elasticapm.label()` instead of `elasticapm.tag()`.
-The `tag()` API will continue to work as before, but emit a `DeprecationWarning`. It will be removed in 6.0 of the agent. 
+The `tag()` API will continue to work as before, but emit a `DeprecationWarning`. It will be removed in 6.0 of the agent.
```
Comment (beniwohli) [inline docs/upgrading.asciidoc:59]: err, did we actually remove `tag()`? :fearful:

Comment (basepi) [reply docs/upgrading.asciidoc:59]: Probably not. Good catch, trailing whitespace!

File: docs/upgrading.asciidoc [hunk 3]
```diff
@@ -25,8 +25,22 @@ The table below is a simplified description of this policy.
...
+=== Upgrading to version 6 of the agent
+
+==== Python 2 no longer supported
+
+Please upgrade to python 3.6+ to continue to receive regular updates.
+
+==== `SANITIZE_FIELD_NAMES` changes
+
+If you are using a non-default `sanitize_field_names` config, please note
+that your entries must be surrounded with stars (i.e. `*secret*`) in order to
```
Comment (bmorelli25) [inline docs/upgrading.asciidoc:41]: It seems like you're listing an example here, so I'd use `e.g.` instead.

```suggestion
that your entries must be surrounded with stars (e.g. `*secret*`) in order to
```

**Review by beniwohli — APPROVED** [review]
awesome turnaround time on the `tag()` removal!

**Pull request merged**